### PR TITLE
[Merged by Bors] - feat: define `sl₂` triples and prove basic result

### PR DESCRIPTION
**Pull request by ocfnash**
---

Split out from #12937 (suggestions there applied here)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/Sl2.lean [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  contrapose! this
+  simpa [this] using t.lie_e_f.symm
+
+variable {R}
+
+variable (h e f) in
+/-- Given a representation of a Lie algebra with distinguished `sl₂` triple, a vector is said to be
+primitive if it is a simultaneous eigenvector for the action of both `h`, `e`, and the eigenvalue
+for `e` is zero. -/
+structure HasPrimitiveVectorWith (m : M) (μ : R) extends IsSl2Triple h e f : Prop where
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Sl2.lean:78]: After a bit more mulling over this, I'm wondering whether `extends` is the right choice. Maybe the triple should be a parameter. That seems to be how you use it in practice anyways. (And then `h e f` could possibly be implicit parameters.)

No strong opinion though.

Comment (ocfnash) [reply Mathlib/Algebra/Lie/Sl2.lean]: Yeah I think I slightly prefer this; implemented in https://github.com/leanprover-community/mathlib4/pull/13076/commits/dc3e7c5a4b8f4800df13b0f831b4c7b0292870cf

File: Mathlib/Algebra/Lie/Sl2.lean [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  ne_zero : m ≠ 0
+  lie_h : ⁅h, m⁆ = μ • m
+  lie_e : ⁅e, m⁆ = 0
+
+/-- Given a representation of a Lie algebra with distinguished `sl₂` triple, a simultaneous
+eigenvector for the action of both `h` and `e` necessarily has eigenvalue zero for `e`. -/
+lemma HasPrimitiveVectorWith.mk' [NoZeroSMulDivisors ℤ M] (t : IsSl2Triple h e f) (m : M) (μ ρ : R)
+    (hm : m ≠ 0) (hm' : ⁅h, m⁆ = μ • m) (he : ⁅e, m⁆ = ρ • m) :
+    HasPrimitiveVectorWith t m μ  where
+  __ := t
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Sl2.lean:87]:
```suggestion
```

**Comment by jcommelin** [comment]
bors d+

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**